dask/tests/test_multiprocessing.py

```diff
@@ -101,3 +104,31 @@ def test_optimize_graph_false():
     with Callback(pretask=lambda key, *args: keys.append(key)):
         get(d, 'z', optimize_graph=False)
     assert len(keys) == 2
+
+
+def py_random():
+    return tuple(random.randint(0, 1000) for i in range(5))
+
+
+def np_random():
+    return tuple(np.random.randint(1000) for i in range(5))
+
+
+def check_random_seed_in_children(func, N=10):
+    """
+    Check that delegating tasks to a process pool doesn't produce
+    the same random sequences in different children.
+    """
+    func = delayed(func, pure=False)
+    with set_options(get=get):
+        results, = compute([func() for i in range(N)])
+
+    assert len(set(results)) == N
+
+
+def test_py_random_seeds():
+    check_random_seed_in_children(py_random)
+
+
+def test_np_random_seeds():
+    check_random_seed_in_children(np_random)
```

Comment:
We might be able to reduce the use of multiple functions here by using pytest parameters.   Mostly my goal here is to avoid the amount of indirection for any piece of code.  This is obviously subjective, but I find it nice to have everything relevant in one place if possible.  I especially appreciate this when debugging and the relevant code is not fresh in my mind.

```python
@pytest.mark.parametrize('random', [np.random, random])
def test_random_seeds(random):
    def f():
        return tuple(random.randint(0, 10000) for i in range(5))

    with set_options(get=get):
        (results,) = compute([delayed(func, pure=False)() for i in range(10)])

    assert len(set(results)) == N
```